packages/data-broker/src/DataBroker.js

```diff
@@ -3,12 +3,19 @@
  * Copyright (c) 2017 - 2020 Beyond Essential Systems Pty Ltd
  */
 
+import groupBy from 'lodash.groupby';
+
+import { countDistinct } from '@tupaia/utils';
 import { createService } from './services';
 import { getModels } from './getModels';
 
 export class DataBroker {
   constructor() {
     this.models = getModels();
+    this.resultMergers = {
+      [this.getDataSourceTypes().DATA_ELEMENT]: this.mergeAnalytics,
+      [this.getDataSourceTypes().DATA_GROUP]: this.mergeEvents,
+    };
```

Comment:
You will notice that `DataBroker` is now aware of data element and event specifics, while in the past it wasn't. `DataBroker` was initially designed to be not only service-agnostic, but also data_source_type-agnostic. The latter doesn't make sense in reality, and is a result of us trying to interpret data elements and data groups as one thing ("data source").

In practice, we always specify the data structure we want to pull, by using either `Aggregator.fetchAnalytics` or `Aggregator.fetchEvents`.